From 8a803276ae7f30a4ec5024be4d35ddaff62da574 Mon Sep 17 00:00:00 2001
From: Abhinav Kishan Potharaju <quic_apothara@quicinc.com>
Date: Tue, 22 Oct 2024 11:24:35 +0530
Subject: [PATCH] Updating 683-of_net-add-mac-address-to-of-tree.patch

[PATCH] of/of_net: write back netdev MAC-address to device-tree

The label-mac logic relies on the mac-address property of a netdev
devices of-node. However, the mac address can also be stored as a
different property or read from e.g. an mtd device.

Create this node when reading a mac-address from OF if it does not
already exist and copy the mac-address used for the device to this
property. This way, the MAC address can be accessed using procfs.

Change-Id: Id645fec195435047186b92cbdbf0b946cfd15491
Signed-off-by: Abhinav Kishan <quic_apothara@quicinc.com>
---
 net/core/of_net.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/core/of_net.c b/net/core/of_net.c
index 1cc80a2dc063..b7824e58f197 100644
--- a/net/core/of_net.c
+++ b/net/core/of_net.c
@@ -204,7 +204,7 @@ int of_get_mac_address(struct device_node *np, u8 *addr)
 		of_remove_property(np, of_find_property(np, "mac-address-increment-byte", NULL));
 	}
 
-	of_add_mac_address(np, addr);
+	ret = of_add_mac_address(np, addr);
 	return ret;
 }
 EXPORT_SYMBOL(of_get_mac_address);
-- 
2.34.1

